Sources/DataPipeline/Config/SDKConfigBuilder.swift

```diff
@@ -0,0 +1,147 @@
+/// SDK is initialized.
+///
+/// Usage Example:
+/// ```
+/// let config = SDKConfigBuilder(writeKey: "your_write_key")
+///   .apiHost("your_api_host")
+///   .flushAt(30)
+///   // additional configuration as needed...
+///   .build()
+/// // Use `config` for initializing the SDK...
+/// ```
+public class SDKConfigBuilder {
+    // default static values for configuration options
+    private static let defaultAPIHost = "cdp.customer.io/v1"
+    private static let defaultCDNHost = "cdp.customer.io/v1"
+
+    // configuration options for SdkConfig
+    private var logLevel: CioLogLevel = .error
+
+    // configuration options for DataPipelineConfigOptions
+    private let writeKey: String
+    private var apiHost: String = SDKConfigBuilder.defaultAPIHost
+    private var cdnHost: String = SDKConfigBuilder.defaultCDNHost
+    private var flushAt: Int = 20
+    private var flushInterval: Seconds = 30
+    private var autoAddCustomerIODestination: Bool = true
+    private var defaultSettings: Settings?
+    private var flushPolicies: [FlushPolicy] = [CountBasedFlushPolicy(), IntervalBasedFlushPolicy()]
+    private var flushQueue: DispatchQueue = .init(label: "com.segment.operatingModeQueue", qos: .utility)
+    private var operatingMode: OperatingMode = .asynchronous
+    private var trackApplicationLifecycleEvents: Bool = true
+    private var autoTrackDeviceAttributes: Bool = true
+
+    // allow construction of builder with required configurations only
+    public init(writeKey: String) {
+        self.writeKey = writeKey
+    }
+
+    @discardableResult
+    public func logLevel(_ logLevel: CioLogLevel) -> SDKConfigBuilder {
+        self.logLevel = logLevel
+        return self
+    }
+
+    @discardableResult
+    public func apiHost(_ apiHost: String) -> SDKConfigBuilder {
+        self.apiHost = apiHost
+        return self
+    }
+
+    @discardableResult
+    public func cdnHost(_ cdnHost: String) -> SDKConfigBuilder {
+        self.cdnHost = cdnHost
+        return self
+    }
+
+    @discardableResult
+    public func flushAt(_ flushAt: Int) -> SDKConfigBuilder {
+        self.flushAt = flushAt
+        return self
+    }
+
+    @discardableResult
+    public func flushInterval(_ flushInterval: Seconds) -> SDKConfigBuilder {
+        self.flushInterval = flushInterval
+        return self
+    }
+
+    @discardableResult
+    public func autoAddCustomerIODestination(_ autoAdd: Bool) -> SDKConfigBuilder {
+        autoAddCustomerIODestination = autoAdd
+        return self
+    }
+
+    @discardableResult
+    public func defaultSettings(_ settings: Settings?) -> SDKConfigBuilder {
+        defaultSettings = settings
+        return self
+    }
+
+    @discardableResult
+    public func flushPolicies(_ policies: [FlushPolicy]) -> SDKConfigBuilder {
+        flushPolicies = policies
+        return self
+    }
+
+    @discardableResult
+    public func addFlushPolicies(_ policies: [FlushPolicy]) -> SDKConfigBuilder {
```

Comment:
This one is additional. I added so customers can append to our flush policies if needed.

Reply:
I wonder if this is confusing to have a `flushPolicies` and a `addFlushPolicies`? I wouldn't want a customer to use the wrong one and they experience weird behavior in their app. 

Makes me want to consider not adding this method to avoid customer confusion. 

I like your idea though.

Reply:
I had the same thoughts while adding this, so I completely agree with you. The idea was to help customers who simply want to append their policy to default values.

But since we both had same concern, I have removed `addFlushPolicies` for now. We can later improve this to help customers if needed.